actions/workflows/st2_chatops_e2e_tests.yaml

```diff
@@ -2,109 +2,140 @@
+        do:
+          - log_ubuntu_version
+      - when: <% succeeded() and result().stdout == '16.04' %>
         publish:
           - websocket:
         do:
-          - setup
-      - when: <% failed() %>
+          - run_with_slack_u16
+      - when: <% succeeded() and result().stdout == '18.04' %>
         publish:
-          - ubuntu_version: <% result().stdout %>
+          - websocket:
         do:
-          - log_ubuntu_version
-          - skip_if_rhel_6
+          - run_with_slack_u18
+      - when: <% failed() or (succeeded() and not ["16.04", "18.04"].contains(result().stdout)) %>
+        do:
+          - switch_on_redhat_version
+
   log_ubuntu_version:
     action: core.echo
     input:
       message: <% ctx().ubuntu_version %>
-  skip_if_rhel_6:
-    action: core.local
+
+  switch_on_redhat_version:
+    action: core.remote
     input:
-      command: "[[ -e /etc/redhat-release ]] && cat /etc/redhat-release 2>/dev/null | sed -r 's/^[a-zA-Z ]*([0-9]){1,}\..*$/\\1/'"
+      hosts: <% ctx().host_ip %>
+      command: "echo -n '\"'; [[ -e /etc/redhat-release ]] && cat /etc/redhat-release 2>/dev/null | sed -r 's/^[a-zA-Z ]*([0-9]){1,}\..*$/\\1/' | tr -d '\\n'; echo '\"'"
     next:
-      - when: <% succeeded() and (int(result().stdout) >= 7) %>
+      - publish:
+          - redhat_version: <% result().stdout %>
+        do:
+          - log_redhat_version
+      - when: <% succeeded() and result().stdout == '7' %>
         publish:
           - websocket: /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
         do:
-          - setup
-      - when: <% failed() %>
-        publish:
-          - redhat_version: <% result().stdout %>
+          - run_with_slack_el7
+      - when: <% failed() or (succeeded() and not ["7"].contains(result().stdout)) %>
         do:
-          - log_redhat_version
-          - noop
+          - log_skipping
+
   log_redhat_version:
     action: core.echo
     input:
       message: <% ctx().redhat_version %>
 
-  # Setup the virtualenv
-  # This is separated from the configure action because it does not (and SHOULD
-  # not be) run with root privileges
-  setup:
-    action: st2cd.st2_chatops_setup_e2e_tests
+  log_skipping:
+    action: core.echo
     input:
-      cwd: <% ctx().cwd %>
-    next:
-      - when: <% succeeded() %>
-        do:
-          - run_with_slack
-      - when: <% failed() %>
-        do:
-          - teardown
-  # Run the tests with Slack
-  run_with_slack:
+      message: "Skipping ChatOps end-to-end tests: <% ctx().ubuntu_version %><% ctx().redhat_version %>"
+
+  # Run the tests with Slack, ST2 IUT on Ubuntu 16.04
+  run_with_slack_u16:
     action: st2cd.st2_chatops_e2e_tests_with_slack
     input:
+      host_ip: <% ctx().host_ip %>
       cwd: <% ctx().cwd %>
-      env:
-        - websocket: <% ctx().websocket %>
+      websocket: <% ctx().websocket %>
+      slack_channel: "{{ st2kv.system.chatops_e2e_tests_slack_channel_u16 }}"
```

Comment:
Let's put a big note somewhere (probably st2tests repo) that Slack bot user needs to be invited to those channel for tests to work.

Reply:
And unlike our old tests - I assume those would break and fail loudly if the bot is not invited to those channels and tests fail because of that :)

Reply:
Yes, these tests are end-to-end, so they will explicitly fail if the bots are not invited to the appropriate channels (and subsequently cannot send or receive messages to/from them).